### PR TITLE
feat: making tf testing github actions for fogg

### PR DESCRIPTION
**Pull request by jakeyheath**

### CODE REVIEW

File: .github/actions/tf-test/action.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
+name: TF Test
```
Comment (jakeyheath) [inline .github/actions/tf-test/action.yml:1]: this one

File: .github/actions/get-tf-version-fogg/action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: Get Terraform Version
```
Comment (jakeyheath) [inline .github/actions/get-tf-version-fogg/action.yml:1]: this one

File: .github/actions/fogg-apply/action.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: Run fogg apply
```
Comment (jakeyheath) [inline .github/actions/fogg-apply/action.yml:1]: this one

File: .github/actions/get-changed-fogg-components/action.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+name: Get Changed Fogg Components
```
Comment (jakeyheath) [inline .github/actions/get-changed-fogg-components/action.yml:1]: this one

File: .github/actions/tf-test/action.yml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  steps:
+    - name: Install SSH key
+      uses: webfactory/ssh-agent@v0.4.1
+      with:
+        ssh-private-key: ${{ inputs.ssh-private-key }}
+    - uses: actions/checkout@v3
+    - name: Config Terraform plugin cache
+      shell: bash
+      run: |
+        echo 'plugin_cache_dir="$HOME/.terraform.d/plugin-cache"' >~/.terraformrc
```
Comment (hspitzley-czi) [inline .github/actions/tf-test/action.yml:36]:
```suggestion
        echo 'plugin_cache_dir="$HOME/.terraform.d/plugin-cache"' > ~/.terraformrc
```

File: .github/actions/tf-test/action.yml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            owner: context.repository.owner,
+            repo: context.repository.repo,
+            issue_number: context.issue.number,
+          })
+          const botComment = comments.find(comment => {
+            return comment.user.type === 'Bot' && comment.body.includes('${{ inputs.working-dir }}')
+          })
+
+          // 2. Prepare format of the comment
+          const output = `#### Terraform Format and Style 🖌\`${{ steps.fmt.outcome }}\ \`${{ inputs.working-dir }}\`
```
Comment (hspitzley-czi) [inline .github/actions/tf-test/action.yml:92]: Where does the `fmt` step come from?

Comment (jakeyheath) [reply .github/actions/tf-test/action.yml:92]: good catch, this is probably a copy/paste bug

**Review by hspitzley-czi — APPROVED** [review]

**Comment by jakeyheath** [comment]
stale

**Pull request closed without merging**